### ISSUE TITLE
[java] UnusedPrivateMethod false-positive with method result

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#1531]
**Affects PMD Version:** 6.10.0

**Rule:** [UnusedPrivateMethod](https://pmd.github.io/pmd-6.10.0/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:** The method is actually used, but PMD doesn't detect this usage. The method argument is the return value of a method call - and the resulting type of this method call is not resolved by PMD yet.

**Code Sample demonstrating the issue:**

```java
public class UnusedPrivateMethodFP {
    private void print(String s) { // <- unused private method?
        System.out.println(s);
    }

    public void run() {
        print(new Integer(1).toString()); // it is used here
    }

    private void print2(String s) {
        System.out.println(s);
    }

    public void run2() {
        String temp = new Integer(1).toString();
        print2(temp); // workaround with extra temporary variable
    }

    private void print3(String s) {
        System.out.println(s);
    }

    public void run3() {
        print3((String)new Integer(1).toString()); // workaround with extra cast
    }
}
```

**Running PMD through:** Maven
